### DISCUSSION TITLE
Key format for client activation

### DISCUSSION THREAD

**Discussion by samapico** [keygen-sh/community#18] [Q&A] [answered]
If I create a new license in the web interface and give it a custom key (e.g. DEMO-123), the actual key ends up being much longer and containing much more data (e.g. `key/REVNTy0xMjM=.suxL5SiFeB5KdZYU[.......................................]WsSDg==`). The API only responds if I use that full key.

I also just noticed that the name of the "Key" field changes to "Embedded data" once a Policy is selected for the license, which makes me think I'm doing something wrong.

My intention is to have a relatively short code that I can send to the client, have them input it in the installer of my software, then validate that key and link it to their machine. All the examples I find seem to use a key that looks like a serial number (e.g. C1B6DE-39A6E3-DE1529-8559A0-4AF593-V3), how can I make my keys with a similar format?

**Comment by ezekg** [comment, marked as answer]
It sounds like you've set [a cryptographic scheme](https://keygen.sh/docs/api/cryptography/#cryptographic-keys) on the policy, which is causing the key to be cryptographically signed. These keys can be verified offline, but are much longer (since they contain a signature of the key's embedded data).

To use a standard license key, you'll want to use a policy that has a `null` scheme.

**Comment by taxilian** [comment]
The other thing that you can do if you still want to use a cryptographic scheme is to use an [activation token](https://keygen.sh/docs/api/licenses/#licenses-relationships-activation-tokens); the activation token might look something like `activ-2d0cbcaa8c0e689c76bb1f60da58c32cv3`; what I'm planning to do (in the process of setting up my system) is to give out the hex portion of that token (`2d0cbcaa8c0e689c76bb1f60da58c32cv3`) as the "activation code" to the customer and then you can call the [/me endpoint](https://keygen.sh/docs/api/profiles/) in order to get the license key associated with that code.

My goal is to have something which can be verified if they are connected to the internet but still work offline, so I wanted a cryptographically verifiable key to store locally, but I don't want the customer to ever have to actually see that key (which is quite long =]). This so far seems like it's going to work well for what I want.
